From 83205c456f5e7635485e3f8ba1614238654ca0b0 Mon Sep 17 00:00:00 2001
From: MengLi <meng.li@windriver.com>
Date: Wed, 6 Mar 2019 13:57:44 +0800
Subject: [PATCH 5242/5242] arch: arm: dts: fix build issues in dts files

After getting patches from sdk kernel, there are a lot of building
Issues caused by dts file. So, modify dts related code from sdk patches
to fix these building issues.

Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx6dl-tx6dl-comtft.dts          |    4 -
 arch/arm/boot/dts/imx6dqscm-1gb-evb-fix-ldo.dts    |    4 +-
 .../imx6dqscm-1gb-evb-interleave-android-ldo.dts   |    2 +-
 arch/arm/boot/dts/imx6q-tx6q-1010-comtft.dts       |    4 -
 arch/arm/boot/dts/imx6q-tx6q-1020-comtft.dts       |    4 -
 arch/arm/boot/dts/imx6qdl-sabresd.dtsi             |    6 -
 arch/arm/boot/dts/imx6qdl-tx6-lcd.dtsi             |   15 -
 arch/arm/boot/dts/imx6sx-sabreauto.dts             |  360 ++++++--------------
 arch/arm/boot/dts/imx6sx-sdb.dtsi                  |   88 +++--
 arch/arm/boot/dts/imx6sx.dtsi                      |    2 +-
 arch/arm/boot/dts/imx6ul-14x14-evk.dtsi            |   20 +-
 arch/arm/boot/dts/imx6ul.dtsi                      |   11 +-
 arch/arm/boot/dts/imx6ull.dtsi                     |    2 +-
 arch/arm/boot/dts/imx7d-sdb-reva-touch.dts         |    2 +-
 arch/arm/boot/dts/imx7d-sdb.dts                    |    4 +-
 arch/arm/boot/dts/imx7d.dtsi                       |    9 +-
 arch/arm/boot/dts/imx7s.dtsi                       |   32 +-
 17 files changed, 173 insertions(+), 396 deletions(-)

diff --git a/arch/arm/boot/dts/imx6dl-tx6dl-comtft.dts b/arch/arm/boot/dts/imx6dl-tx6dl-comtft.dts
index 51a9bb9..ab0ebfb 100644
--- a/arch/arm/boot/dts/imx6dl-tx6dl-comtft.dts
+++ b/arch/arm/boot/dts/imx6dl-tx6dl-comtft.dts
@@ -66,10 +66,6 @@
 	status = "disabled";
 };
 
-&lcd_panel {
-	compatible = "edt,etm0700g0edh6";
-};
-
 &reg_can_xcvr {
 	status = "disabled";
 };
diff --git a/arch/arm/boot/dts/imx6dqscm-1gb-evb-fix-ldo.dts b/arch/arm/boot/dts/imx6dqscm-1gb-evb-fix-ldo.dts
index 1013cf7..fe5d68b 100644
--- a/arch/arm/boot/dts/imx6dqscm-1gb-evb-fix-ldo.dts
+++ b/arch/arm/boot/dts/imx6dqscm-1gb-evb-fix-ldo.dts
@@ -11,7 +11,7 @@
 / {
 	model = "Freescale i.MX6D SCM EVB";
 	compatible = "fsl,imx6q-sabresd", "fsl,imx6q";
-	memory: memory {
+	memory: memory@10000000 {
 		linux,usable-memory = <0x10000000 0x20000000
 				       0x80000000 0x20000000>;
 	};
@@ -52,7 +52,7 @@
 };
 
 &i2c2 {
-	pmic: pfuze100@08 {
+	pmic: pfuze100@8 {
 		regulators {
 			sw1a_reg: sw1ab {
 				regulator-min-microvolt = <300000>;
diff --git a/arch/arm/boot/dts/imx6dqscm-1gb-evb-interleave-android-ldo.dts b/arch/arm/boot/dts/imx6dqscm-1gb-evb-interleave-android-ldo.dts
index 5136cd7..4557072 100644
--- a/arch/arm/boot/dts/imx6dqscm-1gb-evb-interleave-android-ldo.dts
+++ b/arch/arm/boot/dts/imx6dqscm-1gb-evb-interleave-android-ldo.dts
@@ -41,7 +41,7 @@
 };
 
 &i2c2 {
-	pmic: pfuze100@08 {
+	pmic: pfuze100@8 {
 		regulators {
 			sw1a_reg: sw1ab {
 				regulator-min-microvolt = <300000>;
diff --git a/arch/arm/boot/dts/imx6q-tx6q-1010-comtft.dts b/arch/arm/boot/dts/imx6q-tx6q-1010-comtft.dts
index ac3050a..a6ef3fc 100644
--- a/arch/arm/boot/dts/imx6q-tx6q-1010-comtft.dts
+++ b/arch/arm/boot/dts/imx6q-tx6q-1010-comtft.dts
@@ -66,10 +66,6 @@
 	status = "disabled";
 };
 
-&lcd_panel {
-	compatible = "edt,etm0700g0edh6";
-};
-
 &reg_can_xcvr {
 	status = "disabled";
 };
diff --git a/arch/arm/boot/dts/imx6q-tx6q-1020-comtft.dts b/arch/arm/boot/dts/imx6q-tx6q-1020-comtft.dts
index a773f25..103e7fa 100644
--- a/arch/arm/boot/dts/imx6q-tx6q-1020-comtft.dts
+++ b/arch/arm/boot/dts/imx6q-tx6q-1020-comtft.dts
@@ -74,10 +74,6 @@
 	status = "disabled";
 };
 
-&lcd_panel {
-	compatible = "edt,etm0700g0edh6";
-};
-
 &reg_can_xcvr {
 	status = "disabled";
 };
diff --git a/arch/arm/boot/dts/imx6qdl-sabresd.dtsi b/arch/arm/boot/dts/imx6qdl-sabresd.dtsi
index 4886b0b..de10ab4 100644
--- a/arch/arm/boot/dts/imx6qdl-sabresd.dtsi
+++ b/arch/arm/boot/dts/imx6qdl-sabresd.dtsi
@@ -722,12 +722,6 @@
 
 		pinctrl_hdmi_cec: hdmicecgrp {
 			fsl,pins = <
-				MX6QDL_PAD_KEY_ROW2__HDMI_TX_CEC_LINE	0x1f8b0
-			>;
-		};
-
-		pinctrl_hdmi_cec: hdmicecgrp {
-			fsl,pins = <
 				MX6QDL_PAD_KEY_ROW2__HDMI_TX_CEC_LINE 0x108b0
 			>;
 		};
diff --git a/arch/arm/boot/dts/imx6qdl-tx6-lcd.dtsi b/arch/arm/boot/dts/imx6qdl-tx6-lcd.dtsi
index bcd0464..b47e239 100644
--- a/arch/arm/boot/dts/imx6qdl-tx6-lcd.dtsi
+++ b/arch/arm/boot/dts/imx6qdl-tx6-lcd.dtsi
@@ -66,21 +66,6 @@
 		default-brightness-level = <50>;
 	};
 
-	lcd_panel: lcd-panel {
-		compatible = "edt,etm0700g0dh6";
-		pinctrl-names = "default";
-		pinctrl-0 = <&pinctrl_lcd0_pwr>;
-		enable-gpios = <&gpio3 29 GPIO_ACTIVE_HIGH>;
-		power-supply = <&reg_3v3>;
-		backlight = <&backlight>;
-
-		port {
-			lcd_panel_in: endpoint {
-				remote-endpoint = <&lcd_out>;
-			};
-		};
-	};
-
 	lcd@0 {
 		compatible = "fsl,lcd";
 		ipu_id = <0>;
diff --git a/arch/arm/boot/dts/imx6sx-sabreauto.dts b/arch/arm/boot/dts/imx6sx-sabreauto.dts
index 8cd0ba0..bf77d7b 100644
--- a/arch/arm/boot/dts/imx6sx-sabreauto.dts
+++ b/arch/arm/boot/dts/imx6sx-sabreauto.dts
@@ -55,118 +55,123 @@
 		};
 	};
 
-	reg_audio: cs42888_supply {
-		compatible = "regulator-fixed";
-		regulator-name = "cs42888_supply";
-		regulator-min-microvolt = <3300000>;
-		regulator-max-microvolt = <3300000>;
-		regulator-always-on;
-	};
+	regulators {
+		compatible = "simple-bus";
+		#address-cells = <1>;
+		#size-cells = <0>;
 
-	si4763_vio1: vio1_tnr {
-		compatible = "regulator-fixed";
-		regulator-name = "vio1";
-		regulator-min-microvolt = <3300000>;
-		regulator-max-microvolt = <3300000>;
-		regulator-always-on;
-	};
+		reg_audio: cs42888_supply {
+			compatible = "regulator-fixed";
+			regulator-name = "cs42888_supply";
+			regulator-min-microvolt = <3300000>;
+			regulator-max-microvolt = <3300000>;
+			regulator-always-on;
+		};
 
-	si4763_vio2: vio2_tnr {
-		compatible = "regulator-fixed";
-		regulator-name = "vio2";
-		regulator-min-microvolt = <3300000>;
-		regulator-max-microvolt = <3300000>;
-		regulator-always-on;
-	};
+		si4763_vio1: vio1_tnr {
+			compatible = "regulator-fixed";
+			regulator-name = "vio1";
+			regulator-min-microvolt = <3300000>;
+			regulator-max-microvolt = <3300000>;
+			regulator-always-on;
+		};
 
-	si4763_vd: f3v3_tnr {
-		compatible = "regulator-fixed";
-		regulator-name = "vd";
-		regulator-min-microvolt = <3300000>;
-		regulator-max-microvolt = <3300000>;
-		regulator-always-on;
-	};
+		si4763_vio2: vio2_tnr {
+			compatible = "regulator-fixed";
+			regulator-name = "vio2";
+			regulator-min-microvolt = <3300000>;
+			regulator-max-microvolt = <3300000>;
+			regulator-always-on;
+		};
 
-		si4763_va: f5v_tnr {
-		compatible = "regulator-fixed";
-		regulator-name = "va";
-		regulator-min-microvolt = <5000000>;
-		regulator-max-microvolt = <5000000>;
-		regulator-always-on;
-	};
+		si4763_vd: f3v3_tnr {
+			compatible = "regulator-fixed";
+			regulator-name = "vd";
+			regulator-min-microvolt = <3300000>;
+			regulator-max-microvolt = <3300000>;
+			regulator-always-on;
+		};
 
-	vcc_sd3: regulator-vcc-sd3 {
-		compatible = "regulator-fixed";
-		pinctrl-names = "default";
-		pinctrl-0 = <&pinctrl_vcc_sd3>;
-		regulator-name = "VCC_SD3";
-		regulator-min-microvolt = <3000000>;
-		regulator-max-microvolt = <3000000>;
-		gpio = <&gpio2 11 GPIO_ACTIVE_HIGH>;
-		enable-active-high;
-	};
+			si4763_va: f5v_tnr {
+			compatible = "regulator-fixed";
+			regulator-name = "va";
+			regulator-min-microvolt = <5000000>;
+			regulator-max-microvolt = <5000000>;
+			regulator-always-on;
+		};
 
-	reg_usb_otg1_vbus: usb_otg1_vbus {
-		compatible = "regulator-fixed";
-		pinctrl-names = "default";
-		pinctrl-0 = <&pinctrl_usb_otg1_vbus>;
-		regulator-name = "usb_otg1_vbus";
-		regulator-min-microvolt = <5000000>;
-		regulator-max-microvolt = <5000000>;
-		gpio = <&gpio1 9 0>;
-		enable-active-high;
-	};
+		vcc_sd3: regulator-vcc-sd3 {
+			compatible = "regulator-fixed";
+			pinctrl-names = "default";
+			pinctrl-0 = <&pinctrl_vcc_sd3>;
+			regulator-name = "VCC_SD3";
+			regulator-min-microvolt = <3000000>;
+			regulator-max-microvolt = <3000000>;
+			gpio = <&gpio2 11 GPIO_ACTIVE_HIGH>;
+			enable-active-high;
+		};
 
-	reg_usb_otg2_vbus: usb_otg2_vbus {
-		compatible = "regulator-fixed";
-		pinctrl-names = "default";
-		pinctrl-0 = <&pinctrl_usb_otg2_vbus>;
-		regulator-name = "usb_otg2_vbus";
-		regulator-min-microvolt = <5000000>;
-		regulator-max-microvolt = <5000000>;
-		gpio = <&gpio1 12 0>;
-		enable-active-high;
-	};
+		reg_usb_otg1_vbus: usb_otg1_vbus {
+			compatible = "regulator-fixed";
+			pinctrl-names = "default";
+			pinctrl-0 = <&pinctrl_usb_otg1_vbus>;
+			regulator-name = "usb_otg1_vbus";
+			regulator-min-microvolt = <5000000>;
+			regulator-max-microvolt = <5000000>;
+			gpio = <&gpio1 9 0>;
+			enable-active-high;
+		};
 
-	reg_can_wake: regulator@1 {
-		compatible = "regulator-fixed";
-		reg = <1>;
-		regulator-name = "can-wake";
-		regulator-min-microvolt = <3300000>;
-		regulator-max-microvolt = <3300000>;
-		gpio = <&max7310_b 7 GPIO_ACTIVE_HIGH>;
-		enable-active-high;
-	};
+		reg_usb_otg2_vbus: usb_otg2_vbus {
+			compatible = "regulator-fixed";
+			pinctrl-names = "default";
+			pinctrl-0 = <&pinctrl_usb_otg2_vbus>;
+			regulator-name = "usb_otg2_vbus";
+			regulator-min-microvolt = <5000000>;
+			regulator-max-microvolt = <5000000>;
+			gpio = <&gpio1 12 0>;
+			enable-active-high;
+		};
 
-	reg_can_en: regulator@2 {
-		compatible = "regulator-fixed";
-		reg = <2>;
-		regulator-name = "can-en";
-		regulator-min-microvolt = <3300000>;
-		regulator-max-microvolt = <3300000>;
-		gpio = <&max7310_b 5 GPIO_ACTIVE_HIGH>;
-		enable-active-high;
-		vin-supply = <&reg_can_wake>;
-	};
+		reg_can_wake: regulator@1 {
+			compatible = "regulator-fixed";
+			reg = <1>;
+			regulator-name = "can-wake";
+			regulator-min-microvolt = <3300000>;
+			regulator-max-microvolt = <3300000>;
+			gpio = <&max7310_b 7 GPIO_ACTIVE_HIGH>;
+			enable-active-high;
+		};
 
-	reg_can_stby: regulator@3 {
-		compatible = "regulator-fixed";
-		reg = <3>;
-		regulator-name = "can-stby";
-		regulator-min-microvolt = <3300000>;
-		regulator-max-microvolt = <3300000>;
-		gpio = <&max7310_b 4 GPIO_ACTIVE_HIGH>;
-		enable-active-high;
-		vin-supply = <&reg_can_en>;
-	};
+		reg_can_en: regulator@2 {
+			compatible = "regulator-fixed";
+			reg = <2>;
+			regulator-name = "can-en";
+			regulator-min-microvolt = <3300000>;
+			regulator-max-microvolt = <3300000>;
+			gpio = <&max7310_b 5 GPIO_ACTIVE_HIGH>;
+			enable-active-high;
+			vin-supply = <&reg_can_wake>;
+		};
 
-	reg_vref_3v3: regulator@4 {
-		compatible = "regulator-fixed";
-		regulator-name = "vref-3v3";
-		regulator-min-microvolt = <3300000>;
-		regulator-max-microvolt = <3300000>;
-	};
+		reg_can_stby: regulator@3 {
+			compatible = "regulator-fixed";
+			reg = <3>;
+			regulator-name = "can-stby";
+			regulator-min-microvolt = <3300000>;
+			regulator-max-microvolt = <3300000>;
+			gpio = <&max7310_b 4 GPIO_ACTIVE_HIGH>;
+			enable-active-high;
+			vin-supply = <&reg_can_en>;
+		};
 
+		reg_vref_3v3: regulator@4 {
+			compatible = "regulator-fixed";
+			regulator-name = "vref-3v3";
+			regulator-min-microvolt = <3300000>;
+			regulator-max-microvolt = <3300000>;
+		};
+	};
 
 	sound-cs42888 {
 		compatible = "fsl,imx6-sabreauto-cs42888",
@@ -371,7 +376,7 @@
 		out-default = /bits/ 16 <0x1 0x1>;
 	};
 
-	pmic: pfuze100@08 {
+	pmic: pfuze100@8 {
 		compatible = "fsl,pfuze100";
 		reg = <0x08>;
 
@@ -749,12 +754,6 @@
 		>;
 	};
 
-	pinctrl_egalax_int: egalax_intgrp {
-		fsl,pins = <
-			MX6SX_PAD_SD4_RESET_B__GPIO6_IO_22      0x80000000
-		>;
-	};
-
 	pinctrl_enet1_1: enet1grp-1 {
 		fsl,pins = <
 			MX6SX_PAD_ENET1_MDIO__ENET1_MDIO        0xa0b1
@@ -1014,153 +1013,6 @@
 	};
 };
 
-&i2c2 {
-	clock-frequency = <100000>;
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_i2c2>;
-	status = "okay";
-
-	touchscreen@4 {
-		compatible = "eeti,egalax_ts";
-		reg = <0x04>;
-		pinctrl-names = "default";
-		pinctrl-0 = <&pinctrl_egalax_int>;
-		interrupt-parent = <&gpio6>;
-		interrupts = <22 IRQ_TYPE_EDGE_FALLING>;
-		wakeup-gpios = <&gpio6 22 GPIO_ACTIVE_HIGH>;
-	};
-
-	pfuze100: pmic@8 {
-		compatible = "fsl,pfuze100";
-		reg = <0x08>;
-
-		regulators {
-			sw1a_reg: sw1ab {
-				regulator-min-microvolt = <300000>;
-				regulator-max-microvolt = <1875000>;
-				regulator-boot-on;
-				regulator-always-on;
-				regulator-ramp-delay = <6250>;
-			};
-
-			sw1c_reg: sw1c {
-				regulator-min-microvolt = <300000>;
-				regulator-max-microvolt = <1875000>;
-				regulator-boot-on;
-				regulator-always-on;
-				regulator-ramp-delay = <6250>;
-			};
-
-			sw2_reg: sw2 {
-				regulator-min-microvolt = <800000>;
-				regulator-max-microvolt = <3300000>;
-				regulator-boot-on;
-				regulator-always-on;
-			};
-
-			sw3a_reg: sw3a {
-				regulator-min-microvolt = <400000>;
-				regulator-max-microvolt = <1975000>;
-				regulator-boot-on;
-				regulator-always-on;
-			};
-
-			sw3b_reg: sw3b {
-				regulator-min-microvolt = <400000>;
-				regulator-max-microvolt = <1975000>;
-				regulator-boot-on;
-				regulator-always-on;
-			};
-
-			sw4_reg: sw4 {
-				regulator-min-microvolt = <800000>;
-				regulator-max-microvolt = <3300000>;
-				regulator-always-on;
-			};
-
-			swbst_reg: swbst {
-				regulator-min-microvolt = <5000000>;
-				regulator-max-microvolt = <5150000>;
-			};
-
-			snvs_reg: vsnvs {
-				regulator-min-microvolt = <1000000>;
-				regulator-max-microvolt = <3000000>;
-				regulator-boot-on;
-				regulator-always-on;
-			};
-
-			vref_reg: vrefddr {
-				regulator-boot-on;
-				regulator-always-on;
-			};
-
-			vgen1_reg: vgen1 {
-				regulator-min-microvolt = <800000>;
-				regulator-max-microvolt = <1550000>;
-				regulator-always-on;
-			};
-
-			vgen2_reg: vgen2 {
-				regulator-min-microvolt = <800000>;
-				regulator-max-microvolt = <1550000>;
-			};
-
-			vgen3_reg: vgen3 {
-				regulator-min-microvolt = <1800000>;
-				regulator-max-microvolt = <3300000>;
-				regulator-always-on;
-			};
-
-			vgen4_reg: vgen4 {
-				regulator-min-microvolt = <1800000>;
-				regulator-max-microvolt = <3300000>;
-				regulator-always-on;
-			};
-
-			vgen5_reg: vgen5 {
-				regulator-min-microvolt = <1800000>;
-				regulator-max-microvolt = <3300000>;
-				regulator-always-on;
-			};
-
-			vgen6_reg: vgen6 {
-				regulator-min-microvolt = <1800000>;
-				regulator-max-microvolt = <3300000>;
-				regulator-always-on;
-			};
-		};
-	};
-
-	max7322: gpio@68 {
-		compatible = "maxim,max7322";
-		reg = <0x68>;
-		gpio-controller;
-		#gpio-cells = <2>;
-	};
-};
-
-&i2c3 {
-	clock-frequency = <100000>;
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_i2c3>;
-	status = "okay";
-
-	max7310_a: gpio@30 {
-		compatible = "maxim,max7310";
-		reg = <0x30>;
-		gpio-controller;
-		#gpio-cells = <2>;
-	};
-
-	max7310_b: gpio@32 {
-		compatible = "maxim,max7310";
-		reg = <0x32>;
-		gpio-controller;
-		#gpio-cells = <2>;
-	};
-};
-
 &wdog1 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_wdog>;
diff --git a/arch/arm/boot/dts/imx6sx-sdb.dtsi b/arch/arm/boot/dts/imx6sx-sdb.dtsi
index 752953b..260fb5f 100644
--- a/arch/arm/boot/dts/imx6sx-sdb.dtsi
+++ b/arch/arm/boot/dts/imx6sx-sdb.dtsi
@@ -142,14 +142,6 @@
 		gpios = <&gpio2 6 GPIO_ACTIVE_LOW>;
 	};
 
-	reg_vref_3v3: regulator@7 {
-		compatible = "regulator-fixed";
-		reg = <7>;
-		regulator-name = "vref-3v3";
-		regulator-min-microvolt = <3300000>;
-		regulator-max-microvolt = <3300000>;
-	};
-
 	reg_pcie_gpio: regulator-pcie-gpio {
 		compatible = "regulator-fixed";
 		pinctrl-names = "default";
@@ -168,33 +160,49 @@
 		regulator-max-microvolt = <5000000>;
 	};
 
-	reg_pcie: regulator@8 {
-		compatible = "regulator-fixed";
-		reg = <8>;
-		pinctrl-names = "default";
-		pinctrl-0 = <&pinctrl_pcie_reg>;
-		regulator-name = "MPCIE_3V3";
-		regulator-min-microvolt = <3300000>;
-		regulator-max-microvolt = <3300000>;
-		gpio = <&gpio2 1 0>;
-		regulator-always-on;
-		enable-active-high;
-	};
 
-	reg_can_en: regulator@9 {
-		compatible = "regulator-fixed";
-		reg = <9>;
-		regulator-name = "can-en";
-		regulator-min-microvolt = <3300000>;
-		regulator-max-microvolt = <3300000>;
-	};
 
-	reg_can_stby: regulator@10 {
-		compatible = "regulator-fixed";
-		reg = <10>;
-		regulator-name = "can-stby";
-		regulator-min-microvolt = <3300000>;
-		regulator-max-microvolt = <3300000>;
+	regulators {
+		compatible = "simple-bus";
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		reg_vref_3v3: regulator@7 {
+			compatible = "regulator-fixed";
+			reg = <7>;
+			regulator-name = "vref-3v3";
+			regulator-min-microvolt = <3300000>;
+			regulator-max-microvolt = <3300000>;
+		};
+
+		reg_pcie: regulator@8 {
+			compatible = "regulator-fixed";
+			reg = <8>;
+			pinctrl-names = "default";
+			pinctrl-0 = <&pinctrl_pcie_reg>;
+			regulator-name = "MPCIE_3V3";
+			regulator-min-microvolt = <3300000>;
+			regulator-max-microvolt = <3300000>;
+			gpio = <&gpio2 1 0>;
+			regulator-always-on;
+			enable-active-high;
+		};
+
+		reg_can_en: regulator@9 {
+			compatible = "regulator-fixed";
+			reg = <9>;
+			regulator-name = "can-en";
+			regulator-min-microvolt = <3300000>;
+			regulator-max-microvolt = <3300000>;
+		};
+
+		reg_can_stby: regulator@10 {
+			compatible = "regulator-fixed";
+			reg = <10>;
+			regulator-name = "can-stby";
+			regulator-min-microvolt = <3300000>;
+			regulator-max-microvolt = <3300000>;
+		};
 	};
 
 	sound {
@@ -347,7 +355,7 @@
 	pinctrl-0 = <&pinctrl_i2c1>;
 	status = "okay";
 
-	pmic: pfuze100@08 {
+	pmic: pfuze100@8 {
 		compatible = "fsl,pfuze200";
 		reg = <0x08>;
 
@@ -924,18 +932,6 @@
 			>;
 		};
 
-		pinctrl_pcie: pciegrp {
-			fsl,pins = <
-				MX6SX_PAD_ENET1_COL__GPIO2_IO_0 0x10b0
-			>;
-		};
-
-		pinctrl_pcie_reg: pciereggrp {
-			fsl,pins = <
-				MX6SX_PAD_ENET1_CRS__GPIO2_IO_1	0x10b0
-			>;
-		};
-
 		pinctrl_peri_3v3: peri3v3grp {
 			fsl,pins = <
 				MX6SX_PAD_QSPI1A_DATA0__GPIO4_IO_16	0x80000000
diff --git a/arch/arm/boot/dts/imx6sx.dtsi b/arch/arm/boot/dts/imx6sx.dtsi
index 10b5dcf..93f906d 100644
--- a/arch/arm/boot/dts/imx6sx.dtsi
+++ b/arch/arm/boot/dts/imx6sx.dtsi
@@ -920,7 +920,7 @@
 					pd_pci: power-domain@3 {
 						reg = <3>;
 						#power-domain-cells = <0>;
-						power-supply = <&reg_pcie>;
+						power-supply = <&reg_pcie_phy>;
 					};
 
 					pd_disp: power-domain@2 {
diff --git a/arch/arm/boot/dts/imx6ul-14x14-evk.dtsi b/arch/arm/boot/dts/imx6ul-14x14-evk.dtsi
index 066d0bf..f3e8dc0 100644
--- a/arch/arm/boot/dts/imx6ul-14x14-evk.dtsi
+++ b/arch/arm/boot/dts/imx6ul-14x14-evk.dtsi
@@ -51,13 +51,19 @@
 		enable-active-high;
 	};
 
-	reg_can_3v3: regulator@0 {
-		compatible = "regulator-fixed";
-		reg = <0>;
-		regulator-name = "can-3v3";
-		regulator-min-microvolt = <3300000>;
-		regulator-max-microvolt = <3300000>;
-		gpios = <&gpio_spi 3 GPIO_ACTIVE_LOW>;
+	regulators {
+		compatible = "simple-bus";
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		reg_can_3v3: regulator@0 {
+			compatible = "regulator-fixed";
+			reg = <0>;
+			regulator-name = "can-3v3";
+			regulator-min-microvolt = <3300000>;
+			regulator-max-microvolt = <3300000>;
+			gpios = <&gpio_spi 3 GPIO_ACTIVE_LOW>;
+		};
 	};
 
 	reg_gpio_dvfs: regulator-gpio {
diff --git a/arch/arm/boot/dts/imx6ul.dtsi b/arch/arm/boot/dts/imx6ul.dtsi
index 27e7756..687bf37 100644
--- a/arch/arm/boot/dts/imx6ul.dtsi
+++ b/arch/arm/boot/dts/imx6ul.dtsi
@@ -1,7 +1,7 @@
 // SPDX-License-Identifier: GPL-2.0
 //
 // Copyright 2015-2016 Freescale Semiconductor, Inc.
- * Copyright 2017-2018 NXP
+// Copyright 2017-2018 NXP
 
 #include <dt-bindings/clock/imx6ul-clock.h>
 #include <dt-bindings/gpio/gpio.h>
@@ -147,6 +147,7 @@
 		interrupt-parent = <&gpc>;
 		interrupts = <GIC_SPI 49 IRQ_TYPE_LEVEL_HIGH>;
 		fsl,tempmon = <&anatop>;
+		fsl,tempmon-data = <&ocotp>;
 		nvmem-cells = <&tempmon_calib>, <&tempmon_temp_grade>;
 		nvmem-cell-names = "calib", "temp_grade";
 		clocks = <&clks IMX6UL_CLK_PLL3_USB_OTG>;
@@ -746,14 +747,6 @@
 				fsl,anatop = <&anatop>;
 			};
 
-			tempmon: tempmon {
-				compatible = "fsl,imx6ul-tempmon", "fsl,imx6sx-tempmon";
-				interrupts = <GIC_SPI 49 IRQ_TYPE_LEVEL_HIGH>;
-				fsl,tempmon = <&anatop>;
-				fsl,tempmon-data = <&ocotp>;
-				clocks = <&clks IMX6UL_CLK_PLL3_USB_OTG>;
-			};
-
 			caam_snvs: caam-snvs@020cc000 {
 				compatible = "fsl,imx6q-caam-snvs";
 				reg = <0x020cc000 0x4000>;
diff --git a/arch/arm/boot/dts/imx6ull.dtsi b/arch/arm/boot/dts/imx6ull.dtsi
index 88d3f87..308b844 100644
--- a/arch/arm/boot/dts/imx6ull.dtsi
+++ b/arch/arm/boot/dts/imx6ull.dtsi
@@ -761,7 +761,7 @@
 					compatible = "fsl,sec-v4.0-pwrkey";
 					regmap = <&snvs>;
 					interrupts = <GIC_SPI 4 IRQ_TYPE_LEVEL_HIGH>;
-					linux,keycode = <KEY_POWER>;
+					linux,keycode = <116>;
 					wakeup-source;
 				};
 			};
diff --git a/arch/arm/boot/dts/imx7d-sdb-reva-touch.dts b/arch/arm/boot/dts/imx7d-sdb-reva-touch.dts
index d3855e8d..91d37d6 100644
--- a/arch/arm/boot/dts/imx7d-sdb-reva-touch.dts
+++ b/arch/arm/boot/dts/imx7d-sdb-reva-touch.dts
@@ -15,7 +15,7 @@
 	status = "okay";
 
 	tsc2046 {
-		interrupts = <13 0>;
+		interrupts = <GIC_SPI 13 0>;
 		pendown-gpio = <&gpio2 13 0>;
 	};
 };
diff --git a/arch/arm/boot/dts/imx7d-sdb.dts b/arch/arm/boot/dts/imx7d-sdb.dts
index 0a4b56b..9da5e98 100644
--- a/arch/arm/boot/dts/imx7d-sdb.dts
+++ b/arch/arm/boot/dts/imx7d-sdb.dts
@@ -300,7 +300,7 @@
 	clock-frequency = <240000000>;
 	status = "okay";
 	port {
-		mipi_sensor_ep: endpoint1 {
+		mipi_sensor_ep: endpoint@1 {
 			remote-endpoint = <&ov5640_mipi_ep>;
 			data-lanes = <2>;
 			csis-hs-settle = <13>;
@@ -308,7 +308,7 @@
 			csis-wclk;
 		};
 
-		csi_mipi_ep: endpoint2 {
+		csi_mipi_ep: endpoint@2 {
 			remote-endpoint = <&csi_ep>;
 		};
 	};
diff --git a/arch/arm/boot/dts/imx7d.dtsi b/arch/arm/boot/dts/imx7d.dtsi
index 0845fea..c805422 100644
--- a/arch/arm/boot/dts/imx7d.dtsi
+++ b/arch/arm/boot/dts/imx7d.dtsi
@@ -184,14 +184,6 @@
 		status = "okay";
 	};
 
-	tempmon: tempmon {
-		compatible = "fsl,imx7d-tempmon";
-		interrupts = <GIC_SPI 49 IRQ_TYPE_LEVEL_HIGH>;
-		fsl,tempmon =<&anatop>;
-		fsl,tempmon-data = <&ocotp>;
-		clocks = <&clks IMX7D_PLL_SYS_MAIN_CLK>;
-	};
-
 	caam_snvs: caam-snvs@30370000 {
 		compatible = "fsl,imx6q-caam-snvs";
 		reg = <0x30370000 0x10000>;
@@ -384,6 +376,7 @@
 
 	crypto: caam@30900000 {
 		compatible = "fsl,imx7d-caam", "fsl,sec-v4.0";
+		fsl,sec-era = <8>;
 		#address-cells = <1>;
 		#size-cells = <1>;
 		reg = <0x30900000 0x40000>;
diff --git a/arch/arm/boot/dts/imx7s.dtsi b/arch/arm/boot/dts/imx7s.dtsi
index 3adebaf..08ae8c9 100644
--- a/arch/arm/boot/dts/imx7s.dtsi
+++ b/arch/arm/boot/dts/imx7s.dtsi
@@ -140,6 +140,7 @@
 		interrupt-parent = <&gpc>;
 		interrupts = <GIC_SPI 49 IRQ_TYPE_LEVEL_HIGH>;
 		fsl,tempmon =<&anatop>;
+		fsl,tempmon-data = <&ocotp>;
 		nvmem-cells = <&tempmon_calib>,
 			<&tempmon_temp_grade>;
 		nvmem-cell-names = "calib", "temp_grade";
@@ -863,37 +864,6 @@
 				};
 			};
 
-			crypto: caam@30900000 {
-				compatible = "fsl,sec-v4.0";
-				fsl,sec-era = <8>;
-				#address-cells = <1>;
-				#size-cells = <1>;
-				reg = <0x30900000 0x40000>;
-				ranges = <0 0x30900000 0x40000>;
-				interrupts = <GIC_SPI 91 IRQ_TYPE_LEVEL_HIGH>;
-				clocks = <&clks IMX7D_CAAM_CLK>,
-					 <&clks IMX7D_AHB_CHANNEL_ROOT_CLK>;
-				clock-names = "ipg", "aclk";
-
-				sec_jr0: jr0@1000 {
-					compatible = "fsl,sec-v4.0-job-ring";
-					reg = <0x1000 0x1000>;
-					interrupts = <GIC_SPI 105 IRQ_TYPE_LEVEL_HIGH>;
-				};
-
-				sec_jr1: jr1@2000 {
-					compatible = "fsl,sec-v4.0-job-ring";
-					reg = <0x2000 0x1000>;
-					interrupts = <GIC_SPI 106 IRQ_TYPE_LEVEL_HIGH>;
-				};
-
-				sec_jr2: jr1@3000 {
-					compatible = "fsl,sec-v4.0-job-ring";
-					reg = <0x3000 0x1000>;
-					interrupts = <GIC_SPI 114 IRQ_TYPE_LEVEL_HIGH>;
-				};
-			};
-
 			flexcan1: can@30a00000 {
 				compatible = "fsl,imx7d-flexcan", "fsl,imx6q-flexcan";
 				reg = <0x30a00000 0x10000>;
-- 
1.7.9.5

